### PR TITLE
TLS improvements

### LINKED ISSUE #1315: TLS host invalid when using kompose.service.expose with path
When i use 
```kompose.service.expose: "mycluster.local/mypath"
kompose.service.expose.tls-secret: "my-tls-secret"
```

it will generate a ingress.yaml with
```...
tls:
    - hosts: mycluster.local/mypath
```

### PR DESCRIPTION
**Pull request by BoBiene**
fixed #1315

implemented #1316
when ``kompose.service.expose.tls-secret`` is set to ``"true"`` the tls host entry will be genereted

added a github.com/actions build script

### CODE REVIEW

**Comment by BoBiene** [comment]
Hi @sebgoa 
Hi @ngtuna 

i'm not very familiar with Go yet and don't know how to fix the build error in "semaphoreci". 
Can one of you give me a tip?

thanks!

**Comment by BoBiene** [comment]
Hi @kadel,

can you help me out with the failed build on semaphore?

thanks!

/assign @kadel

**Comment by kadel** [comment]
> can you help me out with the failed build on semaphore?

Sure, I'll have a look

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
```
Comment (kadel) [inline .github/workflows/go.yml:25]: This should not be needed at all, kompose is using go.mod will fetch dependencies automatically while building.

**Comment by kadel** [comment]
> > can you help me out with the failed build on semaphore?
> 
> Sure, I'll have a look

Fixed failing build on semaphore CI, but our testing scripts are completely broken. They are failing and it is not correctly reported.
But at least it should not block your PR now.

**Comment by BoBiene** [comment]
> > > can you help me out with the failed build on semaphore?
> > 
> > 
> > Sure, I'll have a look
> 
> Fixed failing build on semaphore CI, but our testing scripts are completely broken. They are failing and it is not correctly reported.
> But at least it should not block your PR now.

Thank you 👍

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: go build -v .
```
Comment (kadel) [inline .github/workflows/go.yml:26]:
```suggestion
      run: make bin
```

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: Test
+      run: go test -v .
```
Comment (kadel) [inline .github/workflows/go.yml:29]:
```suggestion
      run: make test
```

**Comment by BoBiene** [comment]
@kadel I have applied the requested changes, now the tests are failing on github action build: https://github.com/BoBiene/kompose/runs/1066178707?check_suite_focus=true

**Comment by BoBiene** [comment]
@kadel I fixed the gofmt and now the test is successfull:
https://github.com/BoBiene/kompose/runs/1070627908?check_suite_focus=true

**Comment by BoBiene** [comment]
@hangyan / @kadel 
What is still missing here?
Is there anything else I need to do so that we can merge this pull request?

File: pkg/transformer/kubernetes/kubernetes.go [hunk 4]
```diff
@@ -445,15 +445,23 @@ func (k *Kubernetes) initIngress(name string, service kobject.ServiceConfig, por
...
 		}
 	}
-
 	if service.ExposeServiceTLS != "" {
-		ingress.Spec.TLS = []networkingv1beta1.IngressTLS{
-			{
-				Hosts:      hosts,
-				SecretName: service.ExposeServiceTLS,
-			},
+		if service.ExposeServiceTLS != "true" {
```
[the remark this answers is not in the document]
Comment (hangyan) [reply pkg/transformer/kubernetes/kubernetes.go:452]: did'nt kown it is int pending  state for so long... make it a comment now

[the remark this answers is not in the document]
Comment (BoBiene) [reply pkg/transformer/kubernetes/kubernetes.go:452]: @hangyan The idea is to set kompose.service.expose.tls-secret to true to create a https ingress using the default ingress tls certificate. 
I could add a new parameter  - but then the two would be mutually exclusive. You would have to check if tls-secret is not empty and "use-ingress-certificate" is true and then an error would have to be triggered. This way it is much easier.

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+name: Go
```
Comment (hangyan) [inline .github/workflows/go.yml:1]: Is this ci file conflict with the one we already have in master?

Comment (BoBiene) [reply .github/workflows/go.yml:1]: @hangyan This one creates and stores the artifacts the other one did not exist when I created this one. As far as I can see the tests are executed there.

**Comment by hangyan** [comment]
@BoBiene I will merge this now. Sorry for my lazy work on PRs. Thanks for your contribution.

**Pull request merged**